spec/unit/backend/hocon_backend_spec.rb

```diff
@@ -0,0 +1,134 @@
+          raise TypeError
+        end
+        output
+      end
+    end
+
+    describe Hocon_backend do
+      before do
+        Config.load({})
+        Hiera.stubs(:debug)
+        Hiera.stubs(:warn)
+        @cache = FakeCache.new
+        @backend = Hocon_backend.new(@cache)
+      end
+
+      describe "#initialize" do
+        it "should announce its creation" do # because other specs checks this
+          Hiera.expects(:debug).with("Hiera HOCON backend starting")
+          Hocon_backend.new
+        end
+      end
+
+      describe "#lookup" do
+        it "should pick data earliest source that has it for priority searches" do
+          Backend.expects(:datasourcefiles).with(:hocon, {}, "hocon", nil).yields(["one", "/nonexisting/one.hocon"])
+          @cache.value = "key: answer"
+
+          expect(@backend.lookup("key", {}, nil, :priority, nil)).to eq("answer")
+        end
+      end
+
+      describe "handling unexpected HOCON values" do
+          before do
+            Backend.expects(:datasourcefiles).with(:hocon, {}, "hocon", nil).yields(["one", "/nonexisting/one.hocon"])
+          end
+
+          it "throws :no_such_key when key is missing in HOCON" do
+            @cache.value = "{}"
+            expect { @backend.lookup("key", {}, nil, :priority, nil) }.to throw_symbol(:no_such_key)
+          end
+
+          it "returns nil when the HOCON value is nil" do
+            @cache.value = "{key: null}"
+            expect(@backend.lookup("key", {}, nil, :priority, nil)).to be_nil
+          end
+
+          it "throws :no_such_key when the HOCON file is false" do
+            @cache.value = ""
+            expect { @backend.lookup("key", {}, nil, :priority, nil) }.to throw_symbol(:no_such_key)
+          end
+
+          it "raises a TypeError when the HOCON value is not a hash" do
+            @cache.value = "a: [one, two, three]"
+            expect { @backend.lookup("key", {}, nil, :priority, nil) }.to raise_error(TypeError)
+          end
+      end
+
+      it "should build an array of all data sources for array searches" do
+        Backend.expects(:datasourcefiles).with(:hocon, {}, "hocon", nil).multiple_yields(["one", "/nonexisting/one.hocon"], ["two", "/nonexisting/two.hocon"])
+        @cache.expects(:read_file).with("/nonexisting/one.hocon", Hash).returns({"key"=>"answer"})
+        @cache.expects(:read_file).with("/nonexisting/two.hocon", Hash).returns({"key"=>"answer"})
+        expect(@backend.lookup("key", {}, nil, :array, nil)).to eq(["answer", "answer"])
+      end
+
+      it "should ignore empty hash of data sources for hash searches" do
+        Backend.expects(:datasourcefiles).with(:hocon, {}, "hocon", nil).multiple_yields(["one", "/nonexisting/one.hocon"], ["two", "/nonexisting/two.hocon"])
+
+        @cache.expects(:read_file).with("/nonexisting/one.hocon", Hash).returns({})
+        @cache.expects(:read_file).with("/nonexisting/two.hocon", Hash).returns({"key"=>{"a"=>"answer"}})
```

Comment:
These tests are really confusing to me.  It seems like they are mocking `read_file` to make it return a literal Ruby hash.  However, my understanding of the production code is that read_file should return a raw YAML/HOCON string, which the backend should end up parsing via `lookup`.

The YAML tests that you copied this from are doing the same weird mocking here, so this is probably OK/correct, but I'm confused by it and it might be worth digging into just a bit to see if we can better understand what's going on.